legal-api/src/legal_api/resources/business/business_comments.py

```diff
@@ -0,0 +1,114 @@
+# Copyright © 2019 Province of British Columbia
+#
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Searching on a business entity.
+
+Provides all the search and retrieval from the business entity datastore.
+"""
+import datetime
+from http import HTTPStatus
+from typing import Tuple
+
+from flask import g, jsonify, request
+from flask_restplus import Resource, cors
+
+from legal_api.exceptions import BusinessException
+from legal_api.models import Business, Comment, User, db
+from legal_api.services import authorized
+from legal_api.services.comments import validate
+from legal_api.utils.auth import jwt
+from legal_api.utils.util import cors_preflight
+
+from .api_namespace import API
+# noqa: I003; the multiple route decorators cause an erroneous error in line space counting
+
+
+@cors_preflight('GET, POST')
+@API.route('/<string:identifier>/comments', methods=['GET', 'POST', 'OPTIONS'])
+class BusinessCommentResource(Resource):
+    """Business Comment service."""
+
+    @staticmethod
+    @cors.crossdomain(origin='*')
+    @jwt.requires_auth
+    def get(identifier):
+        """Return a JSON object with meta information about the Service."""
+        # basic checks
+        business = Business.find_by_identifier(identifier)
+        err_msg, err_code = BusinessCommentResource._basic_checks(identifier, business, request)
+        if err_msg:
+            return jsonify(err_msg), err_code
+
+        comments = db.session.query(Comment).filter(Comment.business_id == business.id, Comment.filing_id.is_(None))
+
+        rv = []
+        for comment in comments:
+            rv.append(comment.json)
+
+        return jsonify(comments=rv)
+
+    @staticmethod
+    @cors.crossdomain(origin='*')
+    @jwt.requires_auth
+    def post(identifier):
```

Comment:
I have not gone through requirements. Is there a need to edit a comment? In that case PUT should be supported as well

Reply:
no. they cannot edit